src/ResourceManagement/DataLake.Store/Microsoft.Azure.Management.DataLake.Store/Customizations/DataTransfer/DataLakeStoreTransferClient.cs

```diff
@@ -529,6 +539,50 @@ private UploadMetadata GetMetadata()
             }
         }
 
+        private int GetDefaultPerFileThreadCount(long fileLength, long maxSegmentLength)
+        {
+            return (int)Math.Min(Math.Ceiling((double)fileLength / maxSegmentLength), DefaultMaxPerFileThreadCount);
```

Comment:
If the user specifies a max segment length of 0 we will fail earlier than here, but I can add that validation here as well.